travel-agency-service/src/main/java/cz/muni/fi/pa165/bluebat/facade/ReservationFacadeImpl.java

```diff
@@ -67,23 +109,10 @@ public void updateReservation(ReservationDTO reservationDTO) {
         reservation.setPrice(price);
         reservation.setExcursions(new HashSet<>(excursions));
 
-        reservationService.update(reservation);
+        return reservation;
     }
 
-    @Override
-    public void deleteReservation(Long id) {
-        Reservation reservation = reservationService.findById(id);
-        reservationService.delete(reservation);
-    }
-
-    @Override
-    public ReservationDTO getReservationById(Long id) {
-        Reservation reservation = reservationService.findById(id);
-
-        if (reservation == null) {
-            return null;
-        }
-
+    private ReservationDTO mapReservationToReservationDTO(Reservation reservation) {
```

Comment:
Na cviku nám říkal že by jsi neměl pracovat s entitou. Navíc toto by měla dělat i beanMappingService. 

Proto bych si rozmyslel, co by mělo ReservationDTO/ReservationCreateDTO obsahovat, podle mě by mělo stačit CustomerDTO, Long tripId a List<Long> excursionIds, pak by jsi předal service vrstvě při volání create/update Reservation, Trip, Excursions -> a service by to všechno správně spojila, vypočítala cenu a vše správně uložila.

Reply:
No já si myslím, že tady to nevadí, jelikož se jedná o private metodu, která nejde nijak volat zvenku, slouží pouze k tomu, aby se mi ve třídě neopakoval kód. Problém s beanMappingService je ten, že na sebe zvládne namapovat stejné typy, Reservation se s ReservationDTO ale shoduje snad jen IDčkem (Customer a CustomerDTO jsou různé typy, když teda ostatní věci budu držet jen formou IDček, což je dobrý nápad, tak počítám, že to tak stále bude jen v DTO a u entity to budu držet jako objekty). Proto jsem používal tyto mapovací metody, kde jsem to dělal ručně a beanMappingService používal až na mapování těch jednotlivých objektů (jako  třeba CustomerDTO na Customer).

Reply:
No pokud to budeš držet formou IDček, tak pak budeš muset pomocí jiných service získat ty dané entity. Je pravda, že mě nenapadlo, že to jsou vlastně jiné objekt, ale mapování si můžeš nastavit tady: [https://github.com/BaranekD/pa165-blue-bat/blob/main/travel-agency-service/src/main/resources/dozerBeanMapping.xml](url). Natavuje se to nějak takto [http://dozer.sourceforge.net/documentation/mappingclasses.html](url)